asset-transfer-abac/README.md

```diff
@@ -0,0 +1,173 @@
+```
+export RECIPIENT="x509::CN=user1,OU=client,O=Hyperledger,ST=North Carolina,C=US::CN=ca.org1.example.com,O=org1.example.com,L=Durham,ST=North Carolina,C=US"
+peer chaincode invoke $TARGET_TLS_OPTIONS -C mychannel -n abac -c '{"function":"TransferAsset","Args":["Asset1","'"$RECIPIENT"'"]}'
+```
+Query the ledger to verify that the asset has a new owner:
+```
+peer chaincode query -C mychannel -n abac -c '{"function":"ReadAsset","Args":["Asset1"]}'
+```
+We can see that Asset1 with is now owned by User1:
+```
+{"ID":"Asset1","color":"blue","size":20,"owner":"x509::CN=creator1,OU=client+OU=org1,O=Hyperledger,ST=North Carolina,C=US::CN=ca.org1.example.com,O=org1.example.com,L=Durham,ST=North Carolina,C=US","appraisedValue":100}
+```
+
+## Update the asset
+
+Now that the asset has been transferred, the new owner can update the asset properties. The smart contract uses the `GetID()` API to ensure that the update is being submitted by the asset owner. To demonstrate the difference between identity and attribute based access control, lets try to update the asset using the creator1 identity first:
+```
+peer chaincode invoke $TARGET_TLS_OPTIONS -C mychannel -n abac -c '{"function":"UpdateAsset","Args":["Asset1","green","20","100"]}'
+```
+
+Even though creator1 can create new assets, the smart contract detects that the transaction was not submitted by the identity that owns the asset. The command returns the following error:
+```
+Error: endorsement failure during invoke. response: status:500 message:"submitting client not authorized to update asset, does not own asset"
+```
+
+Run the following command to operate as the asset owner by setting the MSP path to User1:
+```
+export CORE_PEER_MSPCONFIGPATH=${PWD}/organizations/peerOrganizations/org1.example.com/users/User1@org1.example.com/msp
+```
+
+We can now update the asset. Run the following command to change the asset color from blue to green. All other aspects of the asset will remain unchanged.
+```
+peer chaincode invoke $TARGET_TLS_OPTIONS -C mychannel -n abac -c '{"function":"UpdateAsset","Args":["Asset1","green","20","100"]}'
+```
+Run the query command again to verify that the asset has changed color:
+```
+peer chaincode query -C mychannel -n abac -c '{"function":"ReadAsset","Args":["Asset1"]}'
+```
+The result will display that Asset1 is now green:
+```
+{"ID":"Asset1","color":"green","size":20,"owner":"x509::CN=creator1,OU=client+OU=org1,O=Hyperledger,ST=North Carolina,C=US::CN=ca.org1.example.com,O=org1.example.com,L=Durham,ST=North Carolina,C=US","appraisedValue":100}
```

Comment:
Owner still appears to be creator1 here.

Reply:
Copy and paste error. Ran through again and fixed.